src/riak_kv_vnode.erl

```diff
@@ -2598,47 +2598,40 @@ non_neg_env(App, EnvVar, Default) when is_integer(Default),
 %% 2. local found, but never acted on this key before (or don't remember it!)
 %% 3. local found, local acted, but incoming has greater count or actor epoch
 %%    This one is tricky, since it indicates some byzantine failure somewhere.
--spec maybe_new_key_epoch(boolean(), #state{},
-                          riak_object:riak_object(),
-                          riak_object:riak_object()) ->
-                                 {binary(), #state{}}.
-maybe_new_key_epoch(false, State, _, _) ->
-    %% Never add a new key epoch when not coordinating
-    %% @TODO (rdb) need to mark actor as dirty though.
-    {State#state.vnodeid, State};
-maybe_new_key_epoch(true, State=#state{counter=#counter_state{use=false}, vnodeid=VId}, _, _) ->
+-spec maybe_new_key_epoch(Coord::boolean(), State::#state{},
+                          LocalObject::riak_object:riak_object(),
+                          IncomingObject::riak_object:riak_object()) ->
+                                 {ActorId:: binary(), #state{}}.
```

Comment:
Ah, I found an existing type `vnodeid()` but I'm not sure if it muddies the water since in some places that means the one generated by the vnode_status_mgr, and some cases it means the actor_id generated by smushing together the vnodeid and epoch. Maybe we need both, using the bitstring <<_:Length>> type spec?